### PR TITLE
Enable engine cache

### PR DESCRIPTION
**Pull request by youbeny**
Enable engine cache

### CODE REVIEW

File: driver/driver.cc [hunk 1]
```diff
@@ -148,6 +148,14 @@ static llvm::cl::opt<int> IpuNum(
     llvm::cl::desc("Num of ipus, should consistent with subgraph num"),
     llvm::cl::init(1), llvm::cl::cat(HaloOptCat));
 
+static llvm::cl::opt<bool> EnableEngineCache(
+    "enable-engine-cache", llvm::cl::desc("Enable popart engine cache"),
```
Comment (weimingzha0) [inline driver/driver.cc:152]: We don't need to modify halo. We should use odla_* APIs from user space.

Comment (youbeny) [reply driver/driver.cc:152]: ok，i found `odla_StoreExecutable/odla_LoadExecutable` is already defined, will implemented in odla_popart.cc, EnableEngineCache option reserved for call halo binary scenario

File: lib/target/generic_cpp/generic_cxx_codegen.cc [hunk 2]
```diff
@@ -532,12 +532,18 @@ static void EmitComputationItems(std::ostream* os, const CXXCodeGenOpts& opts) {
     *os << "bool use_ipu_model = " << opts.use_ipu_model << ";\n";
     *os << "int ipu_num = " << opts.ipu_num << ";\n";
     *os << "int batches_per_step = " << opts.batches_per_step << ";\n";
+    *os << "bool enable_engine_cache = " << opts.enable_engine_cache << ";\n";
+    *os << "const char* cache_dir = " << opts.cache_dir << ";\n";
```
Comment (weimingzha0) [inline lib/target/generic_cpp/generic_cxx_codegen.cc:536]: HALO won't generate such code in future. These will be done by ODLA runtime APIs

Comment (youbeny) [reply lib/target/generic_cpp/generic_cxx_codegen.cc:536]: ok

**Review by weimingzha0 — APPROVED** [review]

**Pull request merged**